src/onnxruntime_benchmark/main.cpp

```diff
@@ -0,0 +1,305 @@
+        model.fill_inputs_outputs_info();
+        auto io_tensors_info = model.get_io_tensors_info();
+        log_model_inputs_outputs(io_tensors_info);
+        std::string target_device = "CPU"; // can be changed when ov provider will be added
+        logger::info << "Device: " << target_device << logger::endl;
+        logger::info << "\tThreads number: " << (FLAGS_nthreads ? std::to_string(FLAGS_nthreads) : "DEFAULT")
+                     << logger::endl;
+
+        log_step(); // Configuring input of the model
+        auto inputs_info = inputs::get_inputs_info(input_files,
+                                                   io_tensors_info.first,
+                                                   FLAGS_layout,
+                                                   FLAGS_shape,
+                                                   FLAGS_mean,
+                                                   FLAGS_scale);
+
+        // determine batch size
+        int batch_size = inputs::get_batch_size(inputs_info);
+        bool is_dynamic_batch = std::any_of(inputs_info.begin(), inputs_info.end(), [](const auto &pair) {
+            const auto &[name, input_descr] = pair;
+            return input_descr.tensor_descr.is_dynamic_batch();
+        });
+        if (FLAGS_b > 0 && is_dynamic_batch) {
+            if (!FLAGS_shape.empty() && batch_size != -1) {
+                logger::warn << "Batch size provided with -b option will be used." << logger::endl;
+            }
+            batch_size = FLAGS_b;
+        }
+        else if (is_dynamic_batch && batch_size == -1) {
+            throw std::logic_error("Model has dynamic batch size, but -b option wasn't provided.");
+        }
+        else if (FLAGS_b > 0) {
+            throw std::logic_error("Can't set batch for model with static batch dimension.");
+        }
+        // setting batch
+        inputs::set_batch_size(inputs_info, batch_size);
+        logger::info << "Set batch to " << batch_size << logger::endl;
+
+        log_step(); // Setting execution parameters
+        // number of inference requests
+        int num_requests = FLAGS_nireq;
+        if (FLAGS_nireq == 0) {
+            num_requests = 1;
+        }
+        // set and align iterations limit
+        int64_t num_iterations = FLAGS_niter;
+        if (num_iterations > 0) {
+            num_iterations = ((num_iterations + num_requests - 1) / num_requests) * num_requests;
+            if (FLAGS_niter != num_iterations) {
+                logger::warn << "Provided number of iterations " << FLAGS_niter << " was changed to " << num_iterations
+                             << " to be aligned with number of inference requests " << num_requests << logger::endl;
+            }
+        }
+        // set time limit
+        uint32_t time_limit_sec = 0;
+        if (FLAGS_t != 0) {
+            time_limit_sec = FLAGS_t;
+        }
+        else if (FLAGS_niter == 0) {
```

Comment:
I configured `.clang-format` to maintain the style when `else` on the next line, it's more readable in case when there are many conditions (in omz we used similar style, except for snake_case).

Reply:
yeah, I started unifying code style in omz back then, but didn't finish. Anyway, it's another project with it's own code style.